### PR TITLE
layer collection listing

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[NamedWorkflows](https://trello.com/c/vVmP3b7a/758-namedworkflows)

File: projects/wave-core-new/src/lib/datasets/add-data/add-data.component.ts [hunk 1]
```diff
@@ -61,6 +62,15 @@ export class AddDataComponent implements OnInit {
         };
     }
 
+    static createLayerCollectionButton(): AddDataButton {
+        return {
+            name: 'Existing Layers',
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/datasets/add-data/add-data.component.ts:67]:
```suggestion
            name: 'Layers',
```

File: projects/wave-core-new/src/lib/layer-collections/layer-collection-list/layer-collection-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    icon(item: LayerCollectionItem): string {
+        return createIconDataUrl(item.type);
+    }
+
+    select(item: LayerCollectionItem): void {
+        if (item.type === 'collection') {
+            this.layoutService.setSidenavContentComponent({
+                component: LayerCollectionListComponent,
+                config: {collection: item.id},
+                keepParent: true,
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/layer-collections/layer-collection-list/layer-collection-list.component.ts:87]: You should remove this, as it de-highlights the current tab.

Instead, you could add a simple "<" | ">" menu where you can go back and forth.
You don't need to focus on the old spot in the list just yet.

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: projects/wave-core-new/src/lib/layer-collections/layer-collection-list/layer-collection-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+    private addLayer(layerName: string, workflowId: UUID): void {
+        this.projectService.getWorkflowMetaData(workflowId).subscribe(
+            (resultDescriptorDict) => {
+                const keys = Object.keys(resultDescriptorDict);
+
+                if (keys.includes('columns')) {
+                    this.addVectorLayer(layerName, workflowId, resultDescriptorDict as VectorResultDescriptorDict);
+                } else if (keys.includes('measurement')) {
+                    this.addRasterLayer(layerName, workflowId, resultDescriptorDict as RasterResultDescriptorDict);
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/layer-collections/layer-collection-list/layer-collection-list.component.ts:103]: I borrowed this from the AddWorkflowComponent. Indeed we should introduce a proper `data_type` field to the workflow meta data

[the remark this answers is not in the document]
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/layer-collections/layer-collection-list/layer-collection-list.component.ts]: I borrowed this from the AddWorkflowComponent. Indeed we should introduce a proper `data_type` field to the workflow meta data

File: projects/wave-core-new/src/lib/datasets/add-data/add-data.component.ts (hunk 1 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/datasets/add-data/add-data.component.ts]: 7e7f6b770db6316b43738aba2264cb2aff7c4770

File: projects/wave-core-new/src/lib/layer-collections/layer-collection-list/layer-collection-list.component.ts (hunk 2 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/layer-collections/layer-collection-list/layer-collection-list.component.ts:90]: so how would I go about this? Add a new `LayerCollectionNavigationComponent` that provides the buttons and a div where the actual list is shown and then an Array of Collection Ids that keep track on the navigation? Or is there an existing Angular component that can help me with this?

Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/layer-collections/layer-collection-list/layer-collection-list.component.ts:90]: Yeah, just put the  "<" | ">" buttons on top and the content below. I guess the thing with the arrays should work.

For the content, you can use a [portal](https://material.angular.io/cdk/portal/overview) to put the listing itself into.

Comment (michaelmattig) [reply projects/wave-core-new/src/lib/layer-collections/layer-collection-list/layer-collection-list.component.ts]: I managed to get the portal working after all: cccc2b7d949e49c7ef04ff378a717339e5d9af26

**Comment by ChristianBeilschmidt** [comment]
bors merge

**Pull request merged**